### ISSUE TITLE
Flow-typed Generic Type causes space-infix-ops fail

### ISSUE THREAD

**Issue by josemigallas** [eslint/eslint#11251]
**Tell us about your environment**
MacOS 10.14.1 (18B75)
VSCode 1.24.1 (1.24.1)

* **ESLint Version:**
`4.19.1`
* **Node Version:**
`8.12.0`
* **npm Version:**
`6.5.0`

**What parser (default, Babel-ESLint, etc.) are you using?**
`"babel-eslint": "^8.2.6"`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "allowImportExportEverywhere": false,
    "codeFrame": false
  },
  "env": {
    "browser": true,
    "node": true,
    "jest": true,
    "jasmine": true
  },
  "globals": {
    "fixture": true,
    "spyOnEvent": true
  },
  "extends": "standard",
  "plugins": [
    "flowtype",
    "react"
  ],
  "rules": {
    "no-duplicate-imports": 0,
    "react/sort-comp": [ 2, {
      "order": [
        "type-annotations",
        "static-methods",
        "life-cycle",
        "everything-else",
        "render"
      ]}
    ],
    "react/no-unused-prop-types": 0,
    "react/jsx-uses-react": 2,
    "react/jsx-uses-vars": 2,
    "react/react-in-jsx-scope": 2,
    "flowtype/boolean-style": [ 2, "boolean" ],
    "flowtype/define-flow-type": 1,
    "flowtype/generic-spacing": [ 2, "never" ],
    "flowtype/no-primitive-constructor-types": 2,
    "flowtype/no-weak-types": 1,
    "flowtype/object-type-delimiter": [ 2, "comma" ],
    "flowtype/require-valid-file-annotation": 2,
    "flowtype/space-after-type-colon": [ 2, "always" ],
    "flowtype/space-before-generic-bracket": [ 2, "never" ],
    "flowtype/space-before-type-colon": [ 2, "never" ],
    "flowtype/union-intersection-spacing": [ 2, "always" ],
    "flowtype/use-flow-type": 1,
    "flowtype/valid-syntax": 1
  },
  "settings": {
    "flowtype": {
      "onlyFilesWithFlowAnnotation": true
    }
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* temp.jsx */

// @flow

type State = {
  +chain: string
}

const initialState: State = {
  chain: ''
}

export const ChainReducer = createReducer<State>(initialState.chain, {
  'ADD_POLICY_TO_CHAIN': () => {}
})
```

```bash
$ eslint -c .eslintrc ./temp.jsx
```

**What did you expect to happen?**
No `space-infix-ops` error since it's a generic type notation.

**What actually happened? Please include the actual, raw output from ESLint.**
```
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration.

/Users/jgallaso/Projects/3scale/porta/app/javascript/src/Policies/reducers/temp.jsx
   5:12  warning  Unexpected use of weak type "any"  flowtype/no-weak-types
  12:55  warning  Unexpected use of weak type "any"  flowtype/no-weak-types
  22:42  error    Infix operators must be spaced     space-infix-ops
  22:48  error    Infix operators must be spaced     space-infix-ops

✖ 4 problems (2 errors, 2 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Are you willing to submit a pull request to fix this bug?**
Not familiar enough with Eslint to do it, I'm afraid.

**Comment by nzakas** [comment]
Thanks for the report. I'm not sure there's much we can do about this because we don't make affordances in the core rules for dialect-specific warnings (as in, because this is a Flow-specific problem, we don't typically try to find a way to work around it, because then we'd need to do the same thing for TypeScript-specific problems and any other dialects that come along).

Your best bet here might just be to use a `/*eslint-disable*/` comment around the offending code, but let's see if others have other ideas.
